File: VB-Examples/02_Data/GoalSeek.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace GoalSeek
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Get the first sheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Set the value of cell A1 to "100"
            sheet.Range("A1").Value = "100"

            ' Define the target cell as A2
            Dim targetCell As CellRange = sheet.Range("A2")

            ' Set the formula of the target cell to calculate the sum of A1 and B1
            targetCell.Formula = "=SUM(A1+B1)"

            ' Define the variable cell as B1
            Dim guessCell As CellRange = sheet.Range("B1")

            ' Create a new GoalSeek object
            Dim goalSeek As New Spire.Xls.GoalSeek()

            ' Try to calculate a trial solution for the goal seek, setting the target cell to 500 and using the guess cell
            Dim result As GoalSeekResult = goalSeek.TryCalculate(targetCell, 500, guessCell)

            ' Determine the solution for the goal seek
            result.Determine()

            ' Save the workbook to a file named "GoalSeek.xlsx" in Excel 2013 format
            workbook.SaveToFile("GoalSeek.xlsx", ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'view the document
            ExcelDocViewer("GoalSeek.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/02_Data/GoalSeek.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace GoalSeek
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            // Set value for cell "A1"
            sheet.Range["A1"].Value = "100";

            // Set formula for cell "A2"
            CellRange targetCell = sheet.Range["A2"];
            targetCell.Formula = "=SUM(A1+B1)";

            // Variable cell
            CellRange gussCell = sheet.Range["B1"];
            Spire.Xls.GoalSeek goalSeek = new Spire.Xls.GoalSeek();

            // Trial solution
            GoalSeekResult result = goalSeek.TryCalculate(targetCell, 500, gussCell);

            //Determine the solution
            result.Determine();

            // Save the file
            workbook.SaveToFile("GoalSeek.xlsx", ExcelVersion.Version2013);

            // Dispose of the workbook object to free up resources
            workbook.Dispose();

            // Launch the document
            ExcelDocViewer("GoalSeek.xlsx");
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
